File: 013/Problem13.mjs
import { performance } from "perf_hooks";
import * as fs from "fs";
import * as readLine from "readline";

var lines = [];
var start;
var end;

function processLineByLine() {
  var lineReader = readLine.createInterface({
    input: fs.createReadStream("numbers.txt"),
  });

  lineReader.on("line", function (line) {
    lines.push(line);
  });

  lineReader.on("close", function () {
    haveLinesInArr();
  });
}

function haveLinesInArr() {
  var num = 0;
  for (var i = 0; i < lines.length; i++) {
    num += parseInt(lines[i]);
  }
  var numString = num.toString();
  var output = "";
  for (var i = 0; i < 11; i++) {
    output += numString[i];
  }
  output = output.replace(".", "");
  console.log(output);
  lastFunction();
}

function lastFunction() {
  end = performance.now();
  console.log("Function took " + (end - start) / 1000);
}

function problem() {
  performance.mark("start");
  var lines = processLineByLine();
}

start = performance.now();
problem();

File: 013/Problem13.java
import java.io.File; // Import the File class
import java.io.FileNotFoundException; // Import this class to handle errors
import java.math.BigInteger;
import java.util.Scanner; // Import the Scanner class to read text files

public class Problem13 {

    public static void main(String[] args) {
        final long startTime = System.nanoTime();
        runProblem();
        final long endTime = System.nanoTime();
        System.out.println("Function took " + (endTime - startTime));
    }

    public static void runProblem() {
        BigInteger num = new BigInteger("0");
        try {
            File gridTxt = new File("./013/numbers.txt");
            Scanner fileReader = new Scanner(gridTxt);
            while (fileReader.hasNextLine()) {
                String line = fileReader.nextLine();
                num = num.add(new BigInteger(line));
            }
            fileReader.close();
        } catch (FileNotFoundException e) {
            System.out.println("An error occurred.");
            e.printStackTrace();

        }
        String numString = num.toString();
        StringBuilder output = new StringBuilder();
        for (int i = 0; i < 10; i++) {
            output.append(numString.charAt(i));
        }
        System.out.println(output);
    }
}
